sqlmesh/migrations/v0034_add_default_catalog.py

```diff
@@ -0,0 +1,332 @@
+"""Add default catalog to snapshots and update names to match new normalization rules."""
+
+from __future__ import annotations
+
+import json
+import typing as t
+
+import pandas as pd
+from sqlglot import exp
+from sqlglot.dialects.dialect import DialectType
+from sqlglot.helper import dict_depth, seq_get
+from sqlglot.optimizer.normalize_identifiers import normalize_identifiers
+
+from sqlmesh.utils.migration import index_text_type
+
+
+def set_default_catalog(
+    table: exp.Table,
+    default_catalog: t.Optional[str],
+) -> exp.Table:
+    if default_catalog and not table.catalog and table.db:
+        table.set("catalog", exp.parse_identifier(default_catalog))
+
+    return table
+
+
+def normalize_model_name(
+    table: str | exp.Table,
+    default_catalog: t.Optional[str],
+    dialect: DialectType = None,
+) -> str:
+    table = exp.to_table(table, dialect=dialect)
+
+    table = set_default_catalog(table, default_catalog)
+    return exp.table_name(normalize_identifiers(table, dialect=dialect), identify=True)
+
+
+def normalize_mapping_schema(mapping_schema: t.Dict, dialect: str) -> t.Dict:
+    normalized_mapping_schema = {}
+    for key, value in mapping_schema.items():
+        if isinstance(value, dict):
+            normalized_mapping_schema[
+                normalize_model_name(key, None, dialect)
```

Comment:
why doesn't this have default catalog?

Reply:
We are normalizing individual keys within this nested dict here and therefore default catalog is irrelevant since just a single name will be passed into normalization. So providing it wouldn't hurt but it also doesn't do anything so I don't include it in the function. 

Added this to help demonstrate:
```
# Example input: {'"catalog"': {'schema': {'table': {'column': 'INT'}}}}
# Example output: {'"catalog"': {'"schema"': {'"table"': {'column': 'INT'}}}}
```